### PR TITLE
(DOCSP-30207): Flutter: Add docs for Core Logging

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-30207

### Staged Changes

- [Logging](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-30207/sdk/flutter/logging/)
- [Sync/Set Sync Log Level](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-30207/sdk/flutter/sync/log-level/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/flutter/logging.txt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+=====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+You can set or change your app's log level to develop or debug your 
+application. You might want to change the log level to log different
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply source/sdk/flutter/logging.txt:14]: Now that I've read the page, I see that we don't actually talk about setting the log level for different develpoment scenarios. If there is no way to do that, then we shouldn't mention it.

File: source/sdk/flutter/sync/log-level.txt [hunk 2]
```diff
@@ -10,6 +10,16 @@ Set the Client Log Level - Flutter SDK
    :depth: 3
    :class: singlecol
 
+.. versionchanged:: 1.1.0
```
Comment (MongoCaleb) [inline source/sdk/flutter/sync/log-level.txt:13]: This page title should now have "Deprecated" added (or, if not, you shouldn't call it deprecated on the other page)

Comment (dacharyc) [reply source/sdk/flutter/sync/log-level.txt]: I talked about this with Kyle earlier today, and we're thinking we should not add "Deprecated" to a page title as long as we don't have versioned docs. This is deprecated in new versions, but if you're still on an older version and looking for info here, it's a valid API and we don't want to mislead with a "Deprecated" in the page title.

I've added info here, though, and have changed the tip to a warning to make it clearer this should not be used in newer versions.

File: source/sdk/flutter/logging.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/flutter/logging.txt:14]: I've added an example for changing the log level to get more detail when trying to debug complex code. Does this address your feedback here?

File: source/sdk/flutter/logging.txt [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+amounts of data depending on the app's environment. You can specify 
+different log levels or custom loggers on a per-isolate basis.
+
+.. seealso::
+
+   This page shows how to set a Realm logger, which was added
+   in Realm Flutter SDK v1.1.0. This supersedes setting the Sync client 
+   log level in earlier versions of the Realm Flutter SDK. For information 
+   on how to set the Sync client log level in an earlier version, refer to
+   :ref:`flutter-client-log-level`.
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/flutter/logging.txt:24]: Yeah, I hear you - it feels weird. But I talked it through with Kyle earlier, and the page itself isn't deprecated - the API is still valid for folks who are on older versions. So I think it's important to surface that here.

Can't wait to clean all this up when we get versioned docs.

**Review by desistefanova — APPROVED** [review]
LGTM. I added a suggestion for explaining the default printing with more details. You can improve my suggestion.

File: source/sdk/flutter/logging.txt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+property from the first isolate:
+
+.. literalinclude:: /examples/generated/flutter/logger.snippet.set-custom-logger.dart
+   :language: dart
+
+This sets the logger for Realm logging in this isolate. If you do not 
+provide a log level, the default is 
+:flutter-sdk:`RealmLogLevel.info <realm/RealmLogLevel/info-constant.html>`.
+Any new spawned isolates that work with Realm get a new ``Realm.logger`` instance, 
+but do not ``print`` by default. You can change the log level per-isolate.
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/sdk/flutter/logging.txt]: This is great info, thanks! I've incorporated these details and added a code snippet showing `Realm.logger.onRecord.listen`.

**Review by MongoCaleb — APPROVED** [review]
LGTM!

**Pull request merged**